### PR TITLE
feat: add missing features to ObjC class

### PR DESCRIPTION
**Pull request by farfromrefug**

### CODE REVIEW

**Comment by farfromrefug** [comment]
I added more features like loading from JSON string and `sizeThatFits`

File: lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift [hunk 1]
```diff
@@ -13,27 +13,65 @@ import UIKit
...
     self.bundle = bundle
+    self.filepath = nil
+    self.json = nil
     super.init()
   }
 
   internal var animation: Animation? {
-    return Animation.named(name, bundle: bundle)
+    if (json != nil) {
+      let jsonData = json?.data(using: .utf8)!
```
Comment (thejohnlima) [inline lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift]: Be careful with `swift force unwrap`.
Would be better doing something like this:
```swift
internal var animation: Animation? {
    if let json = json {
      guard let jsonData = json.data(using: .utf8) else { return nil }
      return try? JSONDecoder().decode(Animation.self, from: jsonData)
    } else if let filepath = filepath {
      return Animation.filepath(filepath)
    } else if let name = name, let bundle = bundle {
      return Animation.named(name, bundle: bundle)
    }
    return nil
  }
```

File: lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift [hunk 2]
```diff
@@ -13,27 +13,65 @@ import UIKit
 /// Use in tandem with CompatibleAnimationView when using Lottie in Objective-C
 @objc
 public final class CompatibleAnimation: NSObject {
-
+  
   @objc
   static func named(_ name: String) -> CompatibleAnimation {
     return CompatibleAnimation(name: name)
   }
-
```
Comment (thejohnlima) [inline lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift]: It is not necessary to remove this break line

File: lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift [hunk 3]
```diff
@@ -109,11 +147,22 @@ public final class CompatibleAnimationView: UIView {
...
   public var realtimeAnimationProgress: CGFloat {
     return animationView.realtimeAnimationProgress
   }
+  
+  @objc
+  public var isAnimationPlaying: Bool {
+    return animationView.isAnimationPlaying
+  }
+  
+
```
Comment (thejohnlima) [inline lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift]: Just one break line is necessary

File: lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift [hunk 4]
```diff
@@ -13,27 +13,65 @@ import UIKit
...
+    } else {
+      return Animation.named(name!, bundle: bundle!)
+    }
   }
 
   // MARK: Private
 
-  private let name: String
-  private let bundle: Bundle
+  private let filepath: String?
```
Comment (thejohnlima) [inline lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift]: Would be better if you move this properties to top

File: lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift [hunk 5]
```diff
@@ -245,6 +294,23 @@ public final class CompatibleAnimationView: UIView {
 
     return UIColor(red: CGFloat(colorValue.r), green: CGFloat(colorValue.g), blue: CGFloat(colorValue.b), alpha: CGFloat(colorValue.a))
   }
+  
+
```
Comment (thejohnlima) [inline lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift]: Just one break line is necessary

File: lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift [hunk 6]
```diff
@@ -245,6 +294,23 @@ public final class CompatibleAnimationView: UIView {
...
+  public func setFloatValue(_ value: CGFloat, forKeypath keypath: CompatibleAnimationKeypath)
+  {
+    
+    let valueProvider = FloatValueProvider(value)
+    animationView.setValueProvider(valueProvider, keypath: keypath.animationKeypath)
+  }
+
+  @objc
+  public func getFloatValue(for keypath: CompatibleAnimationKeypath, atFrame: CGFloat) -> NSNumber?
+  {
```
Comment (thejohnlima) [inline lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift]: I think the default style for opening braces is preceded by a single space and on the same line as the declaration:
```swift
 @objc
  public func setFloatValue(_ value: CGFloat, forKeypath keypath: CompatibleAnimationKeypath) {
    let valueProvider = FloatValueProvider(value)
    animationView.setValueProvider(valueProvider, keypath: keypath.animationKeypath)
  }

  @objc
  public func getFloatValue(for keypath: CompatibleAnimationKeypath, atFrame: CGFloat) -> NSNumber? {
    let value = animationView.getValue(for: keypath.animationKeypath, atFrame: atFrame)
    return value as? NSNumber
  }
```

**Review by boluok — APPROVED** [review]

**Comment by farfromrefug** [comment]
@thejohnlima @boluok could this be merged?

**Pull request closed without merging**